npm/metrics/ipset-inventory.go

```diff
@@ -16,16 +16,14 @@ func GetIPSetInventory(setName string) int {
 	return 0
 }
 
-// SetIPSetInventory sets the number of entries in an IPSet and updates a Prometheus metric.
-func SetIPSetInventory(setName string, val int) {
-	_, exists := ipsetInventoryMap[setName]
-	if exists || val != 0 {
-		ipsetInventoryMap[setName] = val
-		updateIPSetInventory(setName)
-	}
+// RemoveFromIPSetInventory deletes the setName from the IPSetInventory metric.
+// When val is 0, the set will be deleted from the IPSetInventory metric.
+func RemoveFromIPSetInventory(setName string) {
+	ipsetInventoryMap[setName] = 0
+	updateIPSetInventory(setName)
 }
 
-// IncIPSetInventory increases the number of entries in an IPSet and updates a Prometheus metric.
+// IncIPSetInventory increases the number of entries for setName in the IPSetInventory metric.
 func IncIPSetInventory(setName string) {
 	_, exists := ipsetInventoryMap[setName]
```

Comment:
I think we do not need to check `exist`
```go
ipsetInventoryMap[setName]++
```
Also, may consider below since it is also obvious code path, but not strong opinion.
```go
ipsetInventoryMap[setName]++
val := float64(ipsetInventoryMap[setName])
labels := GetIPSetInventoryLabels(setName)		
IPSetInventory.With(labels).Set(val)
```

Reply:
in the current implementation, creating an IPSet doesn't add the setName to the map, so the first IncIPSetInventory(setName) call will add setName to the map

Reply:
I meant below codes are equivalent to `ipsetInventoryMap[setName]++`
```go
        _, exists := ipsetInventoryMap[setName]
	if !exists {
		ipsetInventoryMap[setName] = 0
	}
	ipsetInventoryMap[setName]++
```